automation_tools/__init__.py

```diff
@@ -1026,6 +947,129 @@ def product_install(distribution, create_vm=False, certificate_url=None,
+    execute(enable_satellite_repos, host=host)
+    execute(update_packages, warn_only=True)
+    execute(satellite5_installer, host=host)
+
+
+def satellite5_installer():
+    """Installs Satellite 5 from an ISO image.
+
+    The following environment variables affect this command:
+
+    RHN_USERNAME
+        Red Hat Network username.
+    RHN_PASSWORD
+        Red Hat Network password.
+    ISO_URL
+        The URL where the ISO will be downloaded.
+    SATELLITE_CERT_URL
+        The URL where the activation certificate will be downloaded.
+
+    """
+    iso_url = os.environ.get('ISO_URL')
+    if iso_url is None:
+        print('Please provide a valid URL for the ISO image.')
+        sys.exit(1)
+
+    # Download and mount the ISO
+    print('Downloading ISO...')
+    iso_download(iso_url)
+    run('mkdir -p ISO')
+    run('mount -t iso9660 -o loop *.iso ISO')
+
+    # prepare the answer file
+    opts = {
+        'admin-email': os.environ.get('ADMIN_EMAIL', 'root@localhost'),
+        'rhn-username': os.environ.get('RHN_USERNAME', ''),
+        'rhn-password': os.environ.get('RHN_PASSWORD', ''),
+        'rhn-profile-name': os.environ.get('RHN_PROFILE', ''),
+        'rhn-http-proxy': os.environ.get('RHN_HTTP_PROXY', ''),
+        'rhn-http-proxy-username':
+            os.environ.get('RHN_HTTP_PROXY_USERNAME', ''),
+        'rhn-http-proxy-password':
+            os.environ.get('RHN_HTTP_PROXY_PASSWORD', ''),
+        'ssl-set-org': os.environ.get('SSL_SET_ORG', 'Red Hat'),
+        'ssl-set-org-unit': os.environ.get('SSL_SET_ORG_UNIT', 'Satellite QE'),
+        'ssl-set-city': os.environ.get('SSL_SET_CITY', 'Brno'),
+        'ssl-set-state': os.environ.get('SSL_SET_STATE', 'BRQ'),
+        'ssl-set-country': os.environ.get('SSL_SET_COUNTRY', 'CZ'),
+        'ssl-password': os.environ.get('SSL_PASSWORD', 'reset'),
+        'satellite-cert-url': os.environ.get('SATELLITE_CERT_URL', '')
+    }
+    run(
+        'cat <<EOF > /answers.txt\n'
+        'admin-email={admin-email}\n'
+        'rhn-username={rhn-username}\n'
+        'rhn-password={rhn-password}\n'
+        'rhn-profile-name={rhn-profile-name}\n'
+        'rhn-http-proxy={rhn-http-proxy}\n'
+        'rhn-http-proxy-username={rhn-http-proxy-username}\n'
+        'rhn-http-proxy-password={rhn-http-proxy-password}\n'
+        'ssl-config-sslvhost=y\n'
+        'ssl-set-org={ssl-set-org}\n'
+        'ssl-set-org-unit={ssl-set-org-unit}\n'
+        'ssl-set-city={ssl-set-city}\n'
+        'ssl-set-state={ssl-set-state}\n'
+        'ssl-set-country={ssl-set-country}\n'
+        'ssl-set-email={admin-email}\n'
+        'ssl-password={ssl-password}\n'
+        'satellite-cert-file=/root/SATCERT\n'
+        'enable-tftp=yes\n'
+        'EOF\n'.format(**opts)
+    )
+
+    # download a certificate
+    print('Downloading Certificate...')
+    run('wget -nv -O /root/SATCERT {satellite-cert-url}'.format(**opts))
+    # ...and run the installer script.
+    with cd('/root/ISO'):
```

Comment:
Earlier, the directory `ISO` was created. Here, `/root/ISO` is being used. I would suggest just using `ISO`, especially in case a non-root user executes these instructions. The use of the root user should not be assumed if possible, IMO.